src/main/java/com/ecore/roles/model/Membership.java

```diff
@@ -26,14 +23,17 @@ public class Membership {
     @Type(type = "uuid-char")
     private UUID id;
 
-    @OneToOne
+    @ManyToOne
```

Comment:
This relationship doesn’t allow that one User owns many Roles related.
@ManyToOne is the proper relationship between Membership N-1 Role, then, for each Role we may have multiple Memberships instead. Otherwise @OneToOne you may have only one Membership for each Role.